File: Roadmap/08 - CLASES/python/qv1ko.py
class C1:

    def __init__(self, value = "example"):
        self.c1_string = value

    def get_c1_string(self):
        return self.c1_string

    def set_c1_string(self, value):
        self.c1_string = value

    def __str__(self):
        return f"Class 1 value: {self.c1_string}"


class C2:

    def __init__(self, value = "example"):
        self.c2_string = value

    def get_c2_string(self):
        return self.c2_string

    def set_c2_string(self, value):
        self.c2_string = value

    def __str__(self):
        return f"Class 2 value: {self.c2_string}"


if __name__ == "__main__":

    example1 = C1()
    print(example1)
    example1.set_c1_string("Example 1")
    print(example1.get_c1_string())

    example2 = C2()
    print(example2)
    example2.set_c2_string("Example 2")
    print(example2.get_c2_string())

File: Roadmap/08 - CLASES/typescript/qv1ko.ts
class C1 {

    c1String: string;

    constructor(c1String: string) {
        this.c1String = c1String;
    }

    getC1String() {
        return this.c1String;
    }

    setC1String(c1String: string) {
        this.c1String = c1String;
    }

    toString() {
        return this.c1String;
    }

    print() {
        console.log(this.c1String);
    }

}

class C2 {

    c2String: string;

    constructor(c2String: string) {
        this.c2String = c2String;
    }

    getC2String() {
        return this.c2String;
    }

    setC2String(c2String: string) {
        this.c2String = c2String;
    }

    toString() {
        return this.c2String;
    }

    print() {
        console.log(this.c2String);
    }

}

function main() {
    
    let example1: C1 = new C1("example");
    example1.print();
    example1.setC1String("Example 1");
    console.log(example1.toString());

    let example2: C2 = new C2("example");
    example2.print();
    example2.setC2String("Example 2");
    console.log(example2.toString());

}

main();
